connector/src/main/java/org/jboss/as/connector/subsystems/datasources/XMLDataSourceRuntimeHandler.java

```diff
@@ -307,7 +307,10 @@ else if (attributeName.equals(Constants.EXCEPTION_SORTER_MODULE.getName())) {
             if (dataSource.getValidation().getExceptionSorter() == null) {
                 return;
             }
-            setStringIfNotNull(context, ((ModuleClassLoader)dataSource.getValidation().getExceptionSorter().getClassLoader()).getModule().toString());
+            ModuleClassLoader exceptionSorterClassLoader = (ModuleClassLoader) dataSource.getValidation().getExceptionSorter().getClassLoader();
+            if (exceptionSorterClassLoader != null) {
+                setStringIfNotNull(context, exceptionSorterClassLoader.getModule().toString());
+            }
```

Comment:
For all of these, this code pattern is more robust, as it avoids the NPE but also doesn't assume the classloader type:

```
ClassLoader exceptionSorterClassLoader = dataSource.getValidation().getExceptionSorter().getClassLoader();
if (exceptionSorterClassLoader instanceof ModuleClassLoader) {
   setStringIfNotNull(context, ((ModuleClassLoader)exceptionSorterClassLoader).getModule().toString());
}
```

Reply:
updated